sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/ingestion/models/one/CommonSchemaLogSerializerTest.java

```diff
@@ -129,10 +130,20 @@ public void serializeAndDeserialize() throws JSONException {
         /* Data. */
         log.setData(new Data());
         checkSerialization(serializer, log);
-        log.getData().getProperties().put("a", "b");
-        log.getData().getProperties().put("baseDataType", "custom");
+        log.getData().getProperties().put("baseType", "custom");
         log.getData().getProperties().put("baseData", new JSONObject());
+        log.getData().getProperties().put("a", "b");
         Log copy = serializer.deserializeLog(serializer.serializeLog(log), MockCommonSchemaLog.TYPE);
+
+        /*
+         * equals method not defined in JSONObject so it's comparing strings, which is order sensitive.
+         * Since now Part B is serialized first, it's order sensitive for those 2 keys.
+         * Need to compare Part B, then remove Part B then compare the rest with equals...
```

Comment:
Sounds like we need a unit test for ordering.

Reply:
Interaction based then I guess.